### PR TITLE
[DOC] Update external configuration to include DirectoryConfigProvider

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

### Type of change
**Documentation**

Updates the _ExternalConfiguration schema reference_ section in the _Using_ guide to describe how `DirectoryConfigProvider` can be used instead of `FileConfigProvider`

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
There is nothing wrong per-se with the changes. I just wonder if readers will really understand the difference.

File: documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc [hunk 1]
```diff
@@ -76,13 +77,22 @@ spec:
...
 
 Using volumes instead of environment variables is useful in the following scenarios:
 
 * Mounting truststores or keystores with TLS certificates
 * Mounting a properties file that is used to configure Kafka Connect connectors
 
+Configuration providers load the values from mounted ConfigMaps or Secrets.
+
+* `FileConfigProvider` loads configuration values from properties in a file.
+* `DirectoryConfigProvider` loads configuration values from separate files within a directory structure.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:90]: @scholzj Yes. We can add examples for both.
Something like this?

Secret:

```
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
  labels:
    strimzi.io/kind: KafkaUser
    strimzi.io/cluster: my-cluster
type: Opaque
data:
  ca.crt: # Public key of the client CA
  user.crt: # User certificate that contains the public key of the user
  user.key: # Private key of the user
  user.p12: # PKCS #12 archive file for storing certificates and keys
  user.password: # Password for protecting the PKCS #12 archive file
```
Config for Kafka Connect:

````
  config:
    config.providers: directory
    config.providers.directory.class: org.apache.kafka.common.config.provider.DirectoryConfigProvider
  #...
  externalConfiguration:
    volumes:
      - name: connector-config
        secret:
          secretName: mysecret
````

Connector:

````
kind: KafkaConnector
metadata:
  name: my-source-connector
  labels:
    strimzi.io/cluster: my-connect-cluster
spec:
  class: io.debezium.connector.mysql.MySqlConnector
  tasksMax: 2
  config:
    security.protocol: SSL
    ssl.truststore.type: PEM
    ssl.truststore.location: "${directory:/opt/kafka/external-configuration/connector-config/ca.crt}"
    ssl.keystore.type: PEM
    ssl.keystore.location: ${directory:/opt/kafka/external-configuration/connector-config/user.key}"
    #...
````

[the remark this answers is not in the document]
Comment (tombentley) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:90]: @PaulRMellor your `${directory:...}` should looks like `directory:dir:file`, so I think you need to change the last `/` to a `:`. See the example in https://cwiki.apache.org/confluence/display/KAFKA/KIP-632%3A+Add+DirectoryConfigProvider

[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:90]: @tombentley Thanks Tom

[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:90]: Yeah, that is a nice example.

[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:95]: Thanks. I've added the new example, so we have examples for both providers now. I'v shown the connector config for both.

**Review by scholzj — APPROVED** [review]

File: documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc [hunk 2]
```diff
@@ -76,14 +77,30 @@ spec:
...
-You can also mount ConfigMaps or Secrets to a Kafka Connect pod as volumes.
+Use volumes to mount ConfigMaps or Secrets to a Kafka Connect pod.
 
 Using volumes instead of environment variables is useful in the following scenarios:
 
-* Mounting truststores or keystores with TLS certificates
 * Mounting a properties file that is used to configure Kafka Connect connectors
+* Mounting truststores or keystores with TLS certificates
+
+Configuration providers load the values from mounted ConfigMaps or Secrets.
```
Comment (tombentley) [inline documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:87]: Being pedantic: In general config providers are a way to inject values from outside the configuration. Here we're using the provided config providers to load values from mounted ConfigMaps or Secrets. They could also be loaded from other files in the container, or with a some other config provider from somewhere else entirely.

File: documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc [hunk 3]
```diff
@@ -76,14 +77,30 @@ spec:
...
+
+Volumes are mounted inside the Kafka Connect containers in the path `/opt/kafka/external-configuration/_<volume-name>_`.
```
Comment (tombentley) [inline documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:89]: _at_ the path, or _on_ the path. Or even "using the path ... as the mount point".

**Review by tombentley — APPROVED** [review]
A couple of nits, but LGTM.

File: documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.connect.ExternalConfiguration.adoc:87]: Changed to: _Configuration _providers_ load values from outside the configuration._
Also added: _Use a comma-separated list if you want to add more than one provider, including custom providers.
You can use custom providers to load values from other file locations._

**Review by laidan6000 — APPROVED** [review]

**Pull request merged**